### ISSUE TITLE
error on intellij startup after updating to new version of plugin (4.14.0)

### ISSUE THREAD

**Issue by berinle** [jshiell/checkstyle-idea#142]
Platform: mac os x
IntelliJ Version: 14.1.1
JDK 8

![image](https://cloud.githubusercontent.com/assets/82424/7042529/8acd1a28-ddb0-11e4-8912-ecc46e2ddef4.png)

**Comment by jshiell** [comment]
As per release notes, please use IDEA 14.1.1 with the bundled JDK; otherwise IDEA runs on JDK6 on OS X.

https://download.jetbrains.com/idea/ideaIU-14.1.1-custom-jdk-bundled.dmg

**Comment by berinle** [comment]
Aha, thanks for the fast response. User error ;). As opposed to downloading a brand new version of IDEA, i followed this link https://intellij-support.jetbrains.com/entries/23455956-Selecting-the-JDK-version-the-IDE-will-run-under. You can instruct IDEA which JDK version to start up with.

Basically, I edited `/Applications/IntelliJ\ IDEA\ 14.app/Contents/Info.plist`

and replaced JVMVersion with this instead...
    
    <key>JVMVersion</key>
    <string>1.8*</string>

Everything works now. Thanks!

**Comment by jshiell** [comment]
No worries - I had a private bet how long it'd take until someone reported this one :smile:

You may find the bundled JDK version a bit better - it's a custom JetBrains build of 1.8.0_40 that fixes the sub-pixel font rendering in some setups. Other people are however reporting that scrolling is a bit slow when using it, so if the local JDK works, may as well stick with it.

**Comment by osbald** [comment]
What if you cannot change the JDK we have a company compiled version of Java7, work on a Java8 package isn't planned as yet. Whats the last version of the plugin that will work with Java7 and how do I  replace the non-functional Java8 version IDEA updated to? Failing this I'll have to remove the plugin.

Cannot IDEA plugins not be described by what JDKs they are compatible with? whats the point of updating to an incompatible build? IDEA didn't show me the description of the last plugin changes, just said update these.

**Comment by jshiell** [comment]
You only need to change the IDEA JDK, not your project JDK. Old versions of the plugin can be downloaded from [JetBrains's plugin repo](http://plugins.jetbrains.com/plugin/1065); there's a complete change log on the GitHub project page. You can then swap the plugin in the appropriate place for your OS (~/Library/Application Support/IntelliJIdea14 for OS X, YMMV for other OSs).

AFAIK IDEA offers no facility for specifying language requirements in plugins. It'd be a good feature request to raise with them.

**Comment by osbald** [comment]
Thanks I'll try that. This is a linux VM I have to VNC into for dev work from my laptop, so IDEA 14.1.2  was installed via the .tar.gz archive, so no bundled JDK. IDEA just prompted me to update a number of plugins after the 14.1.2 upgrade. I didn't get any descriptions and clicked yes assuming they were required for compatibility with the new version of IDEA. 

So the older versions should still work with 14.1.2?

**Comment by jshiell** [comment]
I haven't tried them, but there haven't been any SDK changes that should break them. I'd try 4.13.2, pretty sure that should be fine.

**Comment by joshfriend** [comment]
Here's a [link to the Community Edition 14.1.3](
https://download.jetbrains.com/idea/ideaIC-14.1.3-custom-jdk-bundled.dmg) for the people like me that don't use Ultimate...

**Comment by jshiell** [comment]
Thanks. Added next to other link.

**Comment by nickcharles** [comment]
Not sure if I'm missing a step here, but neither of the solutions here work for me. I've tried using the bundled JDK version, I've tried changing the JVMVersion to 1.8*, and I've tried both in conjuction, to no avail.

Any guidance?

**Comment by jshiell** [comment]
It sounds like you're doing the correct thing, so I'm at a bit of a loss as to what's going on.

You can check IDEA is running under the expected JDK by looking at *IntelliJ IDEA* -> *About IntelliJ IDEA*. I'd expect the listed JDK to be 1.8.0_40-internal for the bundled version.

**Comment by nickcharles** [comment]
Looks like no matter what I do, the JRE ends up being 1.6.

Are there other settings that determine the JDK that IntelliJ uses, or is the JVMVersion setting the only thing controlling it?

**Comment by jshiell** [comment]
AFAIK that's it. And if you're using the bundled version, it should all be set up (and is packaged with its own JVM). So I'm rather at a loss as to what's going on, I fear.

**Comment by nickcharles** [comment]
I resolved my issue. I had been using IntelliJ prior to trying to upgrade my JDK to 1.8, so it looks like some preferences were stored somewhere in one of these directories:
https://intellij-support.jetbrains.com/entries/23358108-Directories-used-by-the-IDE-to-store-settings-caches-plugins-and-logs

If I had to guess it was somewhere in either of:
`~/Library/Preferences/<PRODUCT><VERSION>`
`~/Library/Caches/<PRODUCT><VERSION>`

Anyway, I removed IntelliJ from my Applications directory, removed all four of those directories, and then reinstalled IntelliJ using the bundled version. Aside from losing a few preferences that I had set, everything is working now.
